yt_dlp/utils.py

```diff
@@ -5492,7 +5498,7 @@ def traverse_obj(
                             Read as: `[traverse_obj(obj, branch) for branch in branches]`.
         - `function`:       Branch out and return values filtered by the function.
                             Read as: `[value for key, value in obj if function(key, value)]`.
-                            For `Sequence`s, `key` is the index of the value.
+                            For `Iterable`s, `key` is the index of the value.
```

Comment:
As `Iterable` has no `index()`, I chose:
```
                            For `Iterable`s, `key` is the enumeration count of the value.
```

Reply:
While your version is more correct, it's not that this is wrong. I would argue going the way from "index" to "enumerated count" should be obvious when talking about Iterables. Also, since sequences are more common to be used here, it makes sense to refer to it as `index`. Both work imo

Reply:
I don't disagree. I left the `Sequence` text that you changed alone and added the `Iterable` line, making what you describe explicit.